opengrok-restful.el

```diff
@@ -54,18 +54,39 @@
 (defun opengrok-restful-current-line ()
   (buffer-substring-no-properties (line-beginning-position) (line-end-position)))
 
+(defun opengrok-restful-get-file-content (path)
+  (let ((content nil)
+        (download-url (format "%s/raw%s"
+                              (s-replace "/api/v1/search" "" opengrok-restful-url)
+                              path)))
+    (request download-url
+             :type "GET"
+             :headers (list (cons "Authorization" (format "Bearer %s" opengrok-restful-token)))
+             :parser 'buffer-string
+             :sync t
+             :complete (cl-function (lambda (&key data &allow-other-keys)
+                                      (setq content data))))
+    content))
+
 (defun opengrok-restful-jump-to-target (line-content)
   (cl-multiple-value-bind (path linum)
       (save-match-data
         (string-match "/.+:[0-9]+" line-content)
         (split-string (match-string 0 line-content) ":"))
-    (find-file (concat
-                (file-name-as-directory opengrok-restful-source-directory)
-                (substring path 1)))
-    (goto-char (point-min))
-    (forward-line (1- (string-to-number linum)))
-    (kill-buffer opengrok-restful-buffer)
-    ))
+    (let ((content (opengrok-restful-get-file-content path)))
```

Comment:
We can increase a opengrok-restful-read-only-mode option, if option enabled, fetch file from opengrok server, else fallback to find local file